File: src/main/java/net/sublime/rest/mapper/PostMapper.java
package net.sublime.rest.mapper;


import net.sublime.rest.dto.post.PostDTO;
import net.sublime.rest.model.post.Post;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface PostMapper {

    PostDTO toDTO(Post post);

    Post toEntity(PostDTO postDTO);
}

File: src/main/java/net/sublime/rest/mapper/PostMapper.kt
package net.sublime.rest.mapper

import net.sublime.rest.model.post.Post
import net.sublime.rest.dto.post.PostDTO
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface PostMapper {
    fun toDTO(post: Post?): PostDTO
    fun toEntity(postDTO: PostDTO?): Post
}
